File: leetcode-go/Q049_Group_Anagrams/Q049_Group_Anagrams.go
package Q049_Group_Anagrams

import (
	"sort"
	"strings"
)

func groupAnagrams(strs []string) [][]string {
	m := map[string][]string{}
	for _, str := range strs {
		s := strings.Split(str, "")
		sort.Strings(s)
		h := strings.Join(s, "")
		if m[h] == nil {
			m[h] = []string{}
		}
		m[h] = append(m[h], str)
	}
	ans := [][]string{}
	for _, v := range m {
		ans = append(ans, v)
	}
	return ans
}

File: leetcode-java/src/facebook/Q049_Group_Anagrams.java
package facebook;

import java.util.*;

public class Q049_Group_Anagrams {

    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for (String str : strs) {
            char[] ss = str.toCharArray();
            Arrays.sort(ss);
            String key = new String(ss);
            map.computeIfAbsent(key, k -> new LinkedList<>()).add(str);
        }
        return new LinkedList<>(map.values());
    }

}
